crates/apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -449,110 +413,28 @@ where
-                        ) {
-                            // DKG is disabled, txs are not actually encrypted
-                            TxResult {
-                                code: ResultCode::InvalidTx.into(),
-                                info: "The encrypted payload of tx was \
-                                       incorrectly marked as un-decryptable"
-                                    .into(),
-                            }
-                        } else {
-                            match tx.header().expiration {
-                                Some(tx_expiration)
-                                    if block_time > tx_expiration =>
-                                {
-                                    TxResult {
-                                        code: ResultCode::ExpiredDecryptedTx
-                                            .into(),
-                                        info: format!(
-                                            "Tx expired at {:#?}, block time: \
-                                             {:#?}",
-                                            tx_expiration, block_time
-                                        ),
-                                    }
-                                }
-                                _ => TxResult {
-                                    code: ResultCode::Ok.into(),
-                                    info: "Process Proposal accepted this \
-                                           transaction"
-                                        .into(),
-                                },
-                            }
-                        }
-                    }
-                    None => TxResult {
-                        code: ResultCode::ExtraTxs.into(),
-                        info: "Received more decrypted txs than expected"
-                            .into(),
-                    },
-                }
-            }
+            TxType::Decrypted(_) => unreachable!(),
             TxType::Wrapper(wrapper) => {
                 // Account for gas and space. This is done even if the
                 // transaction is later deemed invalid, to
                 // incentivize the proposer to include only
                 // valid transaction and avoid wasting block
                 // resources (ABCI only)
-                let mut tx_gas_meter = TxGasMeter::new(wrapper.gas_limit);
-                if tx_gas_meter.add_wrapper_gas(tx_bytes).is_err() {
-                    // Account for the tx's resources even in case of an error.
-                    // Ignore any allocation error
-                    let _ = metadata
-                        .encrypted_txs_bins
-                        .try_dump(tx_bytes, u64::from(wrapper.gas_limit));
 
+                // Account for the tx's resources even in case of an error.
+                let allocated_gas =
+                    metadata.user_gas.try_dump(u64::from(wrapper.gas_limit));
```

Comment:
Aren't we allocating the size of the tx anymore?

Reply:
yep, check line 224 in `check_proposal_tx`.